### DISCUSSION TITLE
Call for testing: Slint 1.8

### DISCUSSION THREAD

**Discussion by ogoffart** [slint-ui/slint#6206] [Show and tell]
Slint 1.8 will be released in a few days! 🚀 If you're eager to get a head start, you can already try out the upcoming version with our nightly release. We've packed in several new features, improvements, and bug fixes that we think you'll love.

To give it a try, check the [nightly release](https://github.com/slint-ui/slint/releases/tag/nightly) where you'll find all the instructions on how to get started.

A few highlights from the release
  - Property changed callback ([blog](https://slint.dev/blog/property-changed-callback))
  - [Timer](https://snapshots.slint.dev/master/docs/slint/src/language/builtins/elements.html#timer)
  - [SwipeGestureHandler](https://snapshots.slint.dev/master/docs/slint/src/language/builtins/elements.html#swipegesturehandler)
  - UX enhancements in the live preview. (eg on [SlintPad](https://snapshots.slint.dev/master/editor/))
  - More binaries for C++ (for cross-compiling)
  - Math postfix expression on numbers. (eg: `foo.round()` or `bar.mod(10)`)
  - VSCode extension gained a "Slint: Create New Project from Template" command
More in the [ChangeLog](https://github.com/slint-ui/slint/blob/master/CHANGELOG.md)

We'd love to hear your feedback before the final release, so don’t hesitate to share your experience or report any issues you might come across. Your input helps make Slint even better!

**Comment by Enyium** [comment]
With the features `["renderer-skia-opengl"]` in `Cargo.toml`, I first got this error that I didn't get with Slint v1.7.2:

```
error: linking with `link.exe` failed: exit code: 1120
[very long "note" text involving many file paths]
```

Updating Visual Studio Community 2022 with the Visual Studio Installer fixed the problem.

**Reply by ogoffart** [reply]
There was an update of the Skia crate. The updated Skia might need more recent version of Visual Studio. There is a section about this in
https://releases.slint.dev/1.7.2/docs/slint/src/advanced/backends_and_renderers#troubleshooting
(Skia is causing more compilation issues than femtovg, and that's why we haven't made Skia the default renderer)

**Reply by Enyium** [reply]
Maybe the necessity to update in Visual Studio Installer would be worth a mention in the v1.8 release notes.

**Reply by ogoffart** [reply]
Good point. Added a note in the changelog in 34fe6b15372fc907456e3e10f11a87d77a29d4e3

**Comment by yanshay** [comment]
**Timer**
`Timer` is a great addition, works great and very useful !
Along the way I tried 'property changed` event, seems like it is no longer experimentatl? Or maybe I'm behind on that (seen it's experimental in 1.7 in a blog entry from a month ago)? Worth mentioning that as well, it's a great addition as well.

**SwipeGestureHandler**
Also a great addition!
I checked it out on MCU on my application and it worked, but it required a very quick swipe. Not something I'd expect someone to easily do.
It felt quite different than the example on slintpad with a mouse, I didn't try to run the example on MCU yet.
Is there detailed explanation of how exactly it work? There is some explanations in the documentation but it doesn't provide all timeouts, etc.
I believe this is one feature where a lot of the parameters need to be exposed and maybe even introduce more detailed configuration and noise elimination if you target this also for embedded devices where the touch screen is not an iPhone grade touch screen.

**Neovim LSP**
I can't get it to recognize the new features (Timer). It says 'Unknown type Timer'.
At the same time in VSCode it works (upgraded to the nighly plugin).
I did all the upgrade steps mentioned in the nightly release notes.
Is there a nightly `vim-slint` ?
Maybe it will get solved when you officially release?

**Reply by ogoffart** [reply]
Thanks for your feedback. 
We do mention property changed callback in our release notes.

Regarding the SwipeGestureHandler, it seems we will need to provide a way to change the default timeouts or pick better default. Do you think this is something that need to be configured for the whole application, or for each SwipeGestureHandler individually?

As for the vim plugin, I am not familiar with this editor, so i don't know how you can configure it to get the nightly version. That's a question for @hunger

**Reply by yanshay** [reply]
> Regarding the SwipeGestureHandler, it seems we will need to provide a way to change the default timeouts or pick better default. Do you think this is something that need to be configured for the whole application, or for each SwipeGestureHandler individually?

I think it would be enough to allow settings for the entire application. It seems to me more dependent on the device HW rather than application. But if it's easier to implement it on the SwipeGestureHandler rather than share the configuration that's fine as well. You can check if on iPad (so large display) they allow configuring gestures settings, my guess it's hardcoded into the device since they know the exact hw they have, but don't really know.

I also think all parameters need to be exposed, not just timeouts. In the documentation I saw something about 8 logical pixels change triggering a swipe. That number for example should also be configurable because these touch devices are sometimes not very accurate and might require larger threshold. Or resolution might be low and therefore might require smaller number.
